### PR TITLE
Add support for SOUNDTRACK and LOGO to module refs

### PR DESCRIPTION
**Pull request by bcoles**
It's current year

### CODE REVIEW

**Review by wvu — APPROVED** [review]
jenkins test this please

**Comment by wvu-r7** [comment]
Are you sure you don't want to use `Rex::Compat` to display the logo and play the soundtrack? :(

**Comment by busterb** [comment]
Wow, imagine when you 'use' the module the soundtrack starts playing, like when you are scrolling around in a media player or rhythm game. Maybe we'll save that for the Metasploit Kodi plugin.

**Comment by jmartin-r7** [comment]
Jenkins test this please.

**Comment by FireFart** [comment]
we shouldn't support this whole logo and soundtrack for vulns stuff :/

**Comment by bcoles** [comment]
@FireFart  This PR addresses the often unsung heroes of exploit development. A lot of effort goes into finding a bug and crafting an exploit. For too long, the artists behind the bugs have been overlooked. Truly, the ghost writers of our time.

**Comment by busterb** [comment]
Oh, this points to the actual authors of these things, if not the logos themselves. Makes sense.

**Review by wvu — APPROVED** [review]

**Review by wvu — APPROVED** [review]

**Review by wvu — APPROVED** [review]

**Comment by wvu-r7** [comment]
Apparently GitHub reviews can be sent arbitrarily despite no changes being made.

File: lib/msf/base/serializer/readable_text.rb [hunk 1]
```diff
@@ -497,10 +497,16 @@ def self.dump_evasion_options(mod, indent = '')
   def self.dump_references(mod, indent = '')
     output = ''
 
-
     if (mod.respond_to?(:references) && mod.references && mod.references.length > 0)
       output << "References:\n"
 
+      mod.references.each do |ref|
+        if ENV['FUEL_THE_HYPE_MACHINE'] && %w{LOGO SOUNDTRACK}.include?(ref.ctx_id)
```
Comment (wvu-r7) [inline lib/msf/base/serializer/readable_text.rb]: `Rex::Compat.getenv` as per our discussion.

File: lib/msf/base/serializer/readable_text.rb [hunk 2]
```diff
@@ -497,10 +497,16 @@ def self.dump_evasion_options(mod, indent = '')
   def self.dump_references(mod, indent = '')
     output = ''
 
-
     if (mod.respond_to?(:references) && mod.references && mod.references.length > 0)
       output << "References:\n"
 
+      mod.references.each do |ref|
+        if Rex::Compat.getenv('FUEL_THE_HYPE_MACHINE') && %w[LOGO SOUNDTRACK].include?(ref.ctx_id)
```
Comment (wvu-r7) [inline lib/msf/base/serializer/readable_text.rb]: I do like `%w[]` better, seeing as it's an array!

**Pull request merged**